src/Common/Telemetry/Telemetry.php

```diff
@@ -594,11 +594,24 @@ public function calculate_optin_status( $opted = null ) {
 	 * Calculate the optin status for the TEC plugins from various sources.
 	 *
 	 * @since 5.1.1.1
+	 * @since TBD Updated to use filter for onboarding wizard status.
 	 *
 	 * @return bool $show If the modal should show
 	 */
 	public static function calculate_modal_status(): bool {
-		// If we've already opted in, don't show the modal.
+		/**
+		 * Allow plugins to hook in and short-circuit whether the telemetry modal should be shown.
+		 *
+		 * @since TBD
+		 *
+		 * @param bool|null $should_show Whether the telemetry modal should show. Return null to continue with default logic.
+		 */
+		$onboarding_result = apply_filters( 'tec_telemetry_should_show_modal', null );
+		if ( $onboarding_result !== null ) {
+			return $onboarding_result;
```

Comment:
You'll need to make sure to return the proper type (boolean).


```suggestion
			return (bool) $onboarding_result;
```